File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/channel/ChannelsPaginatedOutputModel.kt
package im.api.model.output.channel

import im.api.model.output.PaginationOutputModel
import im.domain.channel.Channel
import im.repository.pagination.Pagination

data class ChannelsPaginatedOutputModel(
    val channels: List<ChannelOutputModel>,
    val pagination: PaginationOutputModel?,
) {
    companion object {
        fun fromDomain(channels: Pagination<Channel>): ChannelsPaginatedOutputModel =
            ChannelsPaginatedOutputModel(
                channels =
                    channels.items.map {
                        ChannelOutputModel
                            .fromDomain(it)
                    },
                pagination = PaginationOutputModel.fromPagination(channels.info),
            )
    }
}

File: code/js/src/Dto/output/channels/ChannelsPaginatedOutputModel.ts
import { ChannelOutputModel } from './ChannelOutputModel';
import { PaginationOutputModel } from '../PaginationOutputModel';

/**
 * Channels paginated output model
 *
 * @param channels The channels.
 * @param pagination The pagination information.
 */
export interface ChannelsPaginatedOutputModel {
    channels: ChannelOutputModel[];
    pagination: PaginationOutputModel;
}
